src/js/providers/tracks-mixin.ts

```diff
@@ -72,19 +79,19 @@ interface TracksMixin extends DefaultProvider {
     getCurrentTextTrack: () => TextTrackLike | void;
     getSubtitlesTrack: () => number;
     addTextTracks: (tracksArray: PlaylistItemTrack[]) => TextTrackLike[];
-    setTextTracks: (tracks: TextTrackList) => void;
-    addTrackListeners: (tracks: TextTrackList) => void;
+    addTrackListeners: (tracks: TextTrackList | TextTrackLike[]) => void;
+    setTextTracks: (tracks: TextTrackList | TextTrackLike[] | null) => void;
```

Comment:
Where do these take `TextTrackLike[]`? 

That is odd esp for `addTrackListeners` because we're using `TextTrackList.addEventListener` - arrays don't have that.